src/ve-utils/application/Application.service.ts

```diff
@@ -68,70 +66,87 @@ export class ApplicationService {
-                (ref: VeRefObject) => {
-                    if (ref.pinnedIds) {
-                        prefs.pinnedIds.push(...ref.pinnedIds)
-                    }
-                    this.cacheSvc.put<VePreferencesObject>(cacheKey, prefs)
-                    deferred.resolve(this.cacheSvc.get<VePreferencesObject>(cacheKey))
-                },
-                (response) => {
-                    deferred.reject(response)
-                }
-            )
+        const cacheKey = this.apiSvc.makeCacheKey({ projectId, refId }, '_hidden_' + projectId + '_settings', false)
+        const cached = this.cacheSvc.get<ProjectSettingsObject>(cacheKey)
+        if (cached && !refresh) {
+            return this.$q.resolve(cached)
         }
-
-        return deferred.promise
+        return new this.$q<ProjectSettingsObject>((resolve, reject) => {
+            this.elementSvc
+                .getElement<ProjectSettingsObject>({
+                    projectId,
+                    refId,
+                    elementId: '_hidden_' + projectId + '_settings',
+                })
+                .then((result) => {
+                    if (result === null) {
+                        this.updateSettings(projectId, refId, null).then(resolve, reject)
+                    } else resolve(result)
+                }, reject)
+        })
     }
 
-    public updatePins = (
+    public createSettings = (
         projectId: string,
         refId: string,
-        prefs: VePreferencesObject
-    ): VePromise<VePreferencesObject, ProjectsResponse> => {
-        const deferred = this.$q.defer<VePreferencesObject>()
-        const cacheKey = this.apiSvc.makeCacheKey({ projectId, refId }, '', false, 'preferences')
-        this.projectSvc.getRef(projectId, refId).then(
-            (ref) => {
-                if (prefs.pinnedIds) {
-                    ref.pinnedIds = prefs.pinnedIds
-                    this.projectSvc.updateRef(ref, projectId).then(
-                        (result: VeRefObject) => {
-                            prefs.pinnedIds = result.pinnedIds
-                            this.cacheSvc.put(cacheKey, prefs)
-                            deferred.resolve(this.cacheSvc.get(cacheKey))
-                        },
-                        (reason) => {
-                            deferred.reject(reason)
-                        }
-                    )
-                } else {
-                    deferred.resolve(prefs)
-                }
-            },
-            (reason) => {
-                deferred.reject(reason)
+        settingsOb?: ProjectSettingsObject
+    ): VePromise<ProjectSettingsObject> => {
+        if (!settingsOb) {
+            settingsOb = {
+                id: '_hidden_' + projectId + '_settings',
+                name: 'View Editor Project Settings',
+                _projectId: projectId,
+                _refId: refId,
+                ownerId: 'holding_bin_' + projectId,
```

Comment:
should have no owner so it doesn't get imported into md project

Reply:
Will it be at the root of the project?